specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/stable/2021-03-07/CommunicationIdentity.json

```diff
@@ -0,0 +1,354 @@
+            "description": "Identifier of the identity.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            }
+          },
+          "204": {
+            "description": "Success"
+          }
+        },
+        "x-ms-examples": {
+          "Revoke access tokens": {
+            "$ref": "./examples/RevokeAccessTokens.json"
+          }
+        }
+      }
+    },
+    "/identities/{id}/:issueAccessToken": {
+      "post": {
+        "tags": [
+          "Token"
+        ],
+        "summary": "Issue a new token for an identity.",
+        "operationId": "CommunicationIdentity_IssueAccessToken",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "description": "Identifier of the identity to issue token for.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Requesting scopes for the new token.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CommunicationIdentityAccessTokenRequest"
+            }
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CommunicationIdentityAccessToken"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Issue an access token": {
+            "$ref": "./examples/IssueAccessToken.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CommunicationIdentityTokenScope": {
+      "description": "List of scopes for an access token.",
+      "enum": [
+        "chat",
+        "voip"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "CommunicationIdentityTokenScope",
+        "modelAsString": true
+      }
+    },
+    "CommunicationIdentityCreateRequest": {
+      "type": "object",
+      "properties": {
+        "createTokenWithScopes": {
+          "description": "Also create access token for the created identity.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationIdentityTokenScope"
+          }
+        }
+      }
+    },
+    "CommunicationError": {
```

Comment:
`CommunicationError` and `CommunicationErrorResponse` are close but not exactly the same as in common.json, https://github.com/Azure/azure-rest-api-specs/blob/master/specification/communication/data-plane/Microsoft.CommunicationServicesCommon/preview/2020-11-19-preview1/common.json

For example, there's a `"type": "object"` missing in the `CommunicationErrorResponse`, `readonly` fields and `x-nullable` and the `innererror` are missing.

Please copy over the latest definitions from common.json.

Reply:
Talked offline, missing x-nullable and x-ms-client-name is okay, we can add them client side via transforms.

Reply:
Can you elaborate on why transforms is the preferred solution (I assume you're referring to an autorest transform directive in the readme)?  I ask because there were a few other projects that relied on this technique and it doesn't really scale; i.e. repeating the transforms for the desired languages plus updating them in case of swagger and/or code generator changes.